### ISSUE TITLE
Block Editor: Add Live Editor functionality

### ISSUE THREAD

**Issue by Misplon** [siteorigin/siteorigin-panels#729]
Add Live Editor functionality to the Block Editor. 

Perhaps other blocks can be partially hidden with a white opacity overlay.

**Comment by AlexGStapleton** [comment]
I investigated adding the complete layout to the live editor view and I found it trickier than expected. Passing the data to the server wasn't overly complicated outside of 

Here are some comments and general thoughts of what needs to be done and any issues I came across:

- The renderer needs to be initially overridden in favor of using [do_blocks](https://developer.wordpress.org/reference/functions/do_blocks/) to render the entire page. The renderer will still need to be able to process SiteOrigin Layout blocks.
- If the same preview method is used for the Classic Editor, [generate_css](https://github.com/siteorigin/siteorigin-panels/blob/develop/inc/renderer.php#L49) needs to be overridden.
- [refresh_panels_data](https://github.com/siteorigin/siteorigin-panels/blob/2.11.0/js/siteorigin-panels/view/live-editor.js#L33) is triggered before the page content locks have been updated so `wp.data.select("core/editor").getEditedPostContent()` can't be used. I couldn't spot any functions/events down the line but I suspect this may be avoidable by getting the data Asynchronously (which would require Babel or a polyfill).
- I stopped investigating before getting where worthwhile with achieving this, but [here's a (slightly) edited version of where live-editor.js](https://gist.githubusercontent.com/AlexGStapleton/e55d6d84bb365be9d8ca19fc129307b8/raw/fbb171f72388ddbf9494e99e0a418544bc6b09b5/live-editor.js) got to. It'll pass the data to the server, but it will still suffer from the aforementioned outdated data.
